helpers/DeployHelper.ts

```diff
@@ -50,44 +55,76 @@ export class DeployHelper {
 
     this.log("deploying verifier...");
 
-    const verifierFactory = await ethers.getContractFactory(verifierContractName);
-    const verifier = await verifierFactory.deploy();
+    let verifier;
+    if (verifierContractName === "VerifierStateTransition") {
+      const verifierDeploy = await ignition.deploy(VerifierStateTransitionModule, {
+        strategy: deployStrategy,
+      });
+      verifier = verifierDeploy.verifierStateTransition;
+    } else if (verifierContractName === "VerifierStub") {
+      const verifierDeploy = await ignition.deploy(VerifierStubModule, {
+        strategy: deployStrategy,
+      });
+      verifier = verifierDeploy.verifierStub;
+    } else {
+      throw new Error("invalid verifierContractName");
+    }
     await verifier.waitForDeployment();
     this.log(
-      `${verifierContractName} contract deployed to address ${await verifier.getAddress()} from ${await owner.getAddress()}`
+      `${verifierContractName} contract deployed to address ${await verifier.getAddress()} from ${await owner.getAddress()}`,
     );
 
     this.log("deploying poseidons...");
-    const [poseidon1Elements, poseidon2Elements, poseidon3Elements, poseidon4Elements] = await deployPoseidons(
-      owner,
-      [1, 2, 3, 4]
-    );
+    const [poseidon1Elements, poseidon2Elements, poseidon3Elements, poseidon4Elements] =
+      await deployPoseidons([1, 2, 3, 4], deployStrategy);
 
     this.log("deploying SmtLib...");
     const smtLib = await this.deploySmtLib(
       await poseidon2Elements.getAddress(),
       await poseidon3Elements.getAddress(),
+      "SmtLib",
+      deployStrategy,
     );
 
     this.log("deploying StateLib...");
-    const stateLib = await this.deployStateLib();
+    const stateLib = await this.deployStateLib(deployStrategy);
 
     this.log("deploying state...");
-    const StateFactory = await ethers.getContractFactory("State", {
-      libraries: {
-        StateLib: await stateLib.getAddress(),
-        SmtLib: await smtLib.getAddress(),
-        PoseidonUnit1L: await poseidon1Elements.getAddress(),
-      },
-    });
-    const state = await upgrades.deployProxy(
-      StateFactory,
-      [await verifier.getAddress(), defaultIdType, await owner.getAddress()],
-      {
-        unsafeAllowLinkedLibraries: true,
-    });
+    let state;
+    if (useOpenzeppelinPlugin) {
```

Comment:
can it depend on stratege and not on another variable?

Reply:
yes, refactored to use `deployStrategy` var